Percolation.java

```diff
@@ -1,14 +1,76 @@
+
 import edu.princeton.cs.algs4.StdRandom;
 import edu.princeton.cs.algs4.StdStats;
-import edu.princeton.cs.algs4.WeightedQuickUnionUF;
-
-public class Percolation {
-   public Percolation(int n)                // create n-by-n grid, with all sites blocked
-   public    void open(int row, int col)    // open site (row, col) if it is not open already
-   public boolean isOpen(int row, int col)  // is site (row, col) open?
-   public boolean isFull(int row, int col)  // is site (row, col) full?
-   public     int numberOfOpenSites()       // number of open sites
-   public boolean percolates()              // does the system percolate?
-
-   public static void main(String[] args)   // test client (optional)
-}
\ No newline at end of file
+import princeton.cs.algs4.WeightedQuickUnionUF;
+
+
+ // given an n-by-n matrix of open sites, return an n-by-n matrix
+    // of sites reachable from the top
+  public boolean isOpen(int row, int col){
+    //public static boolean[n][n] flow(boolean[][] isOpen) {
+        string  n =is0pen.length();
+        boolean[][] isFull = new boolean[n][n];
+        for (int j = 0; j < n; j++) {
+            flow(isOpen, isFull, 0, j);
+        }
+        return isFull;
+    }
+// determine set of full sites using depth first search
+    public static void flow(boolean[][] isOpen, boolean[][] isFull, int i, int j) {
+        int n = isOpen.length;
+
+        // base cases
+        if (i < 0 || i >= n) return;    // invalid row
+        if (j < 0 || j >= n) return;    // invalid column
+        if (!isOpen[i][j]) return;      // not an open site
+        if (isFull[i][j]) return;       // already marked as full
+
+        // mark i-j as full
+        isFull[i][j] = true;
+
+        flow(isOpen, isFull, i+1, j);   // down
+        flow(isOpen, isFull, i, j+1);   // right 
+        flow(isOpen, isFull, i, j-1);   // left
+        flow(isOpen, isFull, i-1, j);   // up
+    }
+
+
+    // does the system percolate?
+    public static boolean percolates(boolean[][] isOpen) {
+        int n = isOpen.length;
+        boolean[][] isFull = flow(isOpen);
+        for (int j = 0; j < n; j++) {
+            if (isFull[n-1][j]) return true;
+        }
```

Comment:
If the isFull was actually returning full sites, then your method would be working. However, it would have not respected performance requirements, as this check is O(n) not constant and flow is not constant also.